src/python/grpcio_tests/tests/unit/framework/common/__init__.py

```diff
@@ -11,3 +11,66 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import contextlib
+import socket
+
+
+def get_socket(bind_address='localhost',
+               listen=True,
+               sock_options=(socket.SO_REUSEPORT,)):
```

Comment:
Most implementation looks good. Maybe we want to disable `SO_REUSEPORT` instead of enable it. Existing test server is explicitly disabling that option, so it won't conflict with other test cases.

Reply:
I think `SO_REUSEPORT` is actually appropriate in the cases where we're using this to reserve a port. If there isn't a period where this socket and the server's socket are bound to the same port, then there exists a race condition between the test server and other processes on the machine. I suggest that we simply limit usage of SO_REUSEPORT to those tests that actually need it. If we manage our ports properly, then crosstalk will not be an issue.